components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/filter/CaptchaFilter.java

```diff
@@ -110,7 +113,13 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
             HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
             HttpServletResponse httpResponse = (HttpServletResponse) servletResponse;
 
-            String redirectURL = FrameworkUtils.getContextData(httpRequest).getRedirectURL();
+            AuthenticationContext authenticationContext = FrameworkUtils.getContextData(httpRequest);
+
+            if (authenticationContext != null) {
+                redirectURL = FrameworkUtils.getContextData(httpRequest).getRedirectURL();
```

Comment:
Isn't FrameworkUtils.getContextData(httpRequest) part redundant?
shouldn't this be authenticationContext.getRedirectURL()

Reply:
Yes, it is redundant and resolved now.